### PR TITLE
Add data catalogue shir config

### PR DESCRIPTION
**Pull request by HarrisonBoyleThomas**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/DC-78

### Change description ###
Add an app to run a Purview SHIR for the Data Catalogue project

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

**Comment by HarrisonBoyleThomas** [comment]
We haven't created a stg or prod environment for our project yet - can we go ahead with merging this so we can test it in dev?

**Review by timja — CHANGES_REQUESTED** [review]
please add to just dev first and test there, don't add to prod in the same PR

**Comment by HarrisonBoyleThomas** [comment]
@timja I've removed the prod files, ready for rereview. Also are you able to grant `data-catalogue-team` write access to the repo pls?

**Comment by timja** [comment]
added the team with access

File: CODEOWNERS [hunk 1]
```diff
@@ -54,8 +54,13 @@ apps/darts-modernisation/ @hmcts/darts
 ### JUROR
 apps/juror/ @hmcts/juror
 
+<<<<<<< HEAD
+### DATA CATALOGUE (DC)
+apps/dc/ @hmcts/data-catalogue-team
+=======
 ### OPAL
 apps/opal/ @hmcts/opal
+>>>>>>> master
```
Comment (timja) [inline CODEOWNERS:63]:
```suggestion

### DATA CATALOGUE (DC)
apps/dc/ @hmcts/data-catalogue-team

### OPAL
apps/opal/ @hmcts/opal

```

File: CODEOWNERS [hunk 2]
```diff
@@ -54,9 +54,14 @@ apps/darts-modernisation/ @hmcts/darts
...
 apps/juror/ @hmcts/juror
 
+
+### DATA CATALOGUE (DC)
+apps/dc/ @hmcts/data-catalogue-team
+
 ### OPAL
 apps/opal/ @hmcts/opal
 
+
```
Comment (timja) [inline CODEOWNERS:64]:
```suggestion
```

File: CODEOWNERS [hunk 3]
```diff
@@ -54,9 +54,14 @@ apps/darts-modernisation/ @hmcts/darts
 ### JUROR
 apps/juror/ @hmcts/juror
 
+
```
Comment (timja) [inline CODEOWNERS:57]:
```suggestion
```

File: apps/dc/dc-purview-shir/dc-purview-shir.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+metadata:
+  name: dc-purview-shir-deployment
+  namespace: dc
+spec:
+  interval: 5m
+  releaseName: dc-purview-shir
+  values:
+    labels:
+      app.kubernetes.io/instance: dc-purview-shir-deployment
+      app.kubernetes.io/name: dc-purview-shir-deployment
```
Comment (timja) [inline apps/dc/dc-purview-shir/dc-purview-shir.yaml:12]: I don't think this is supported but should just be added automatically so remove
```suggestion
```

File: apps/dc/dc-purview-shir/dc-purview-shir.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    image: sdshmctspublic.azurecr.io/dc/purview-integration-runtime:prod-844cca3-20230821092242 #{"$imagepolicy": "flux-system:dc-purview-shir"}
+    keyVaults:
+      dc-vault-prod:
+        excludeEnvironmentSuffix: true
```
Comment (timja) [inline apps/dc/dc-purview-shir/dc-purview-shir.yaml:16]: seems wrong?
```suggestion
      dc-vault:
```

File: CODEOWNERS (hunk 1 above)
Comment (HarrisonBoyleThomas) [reply CODEOWNERS]: Sorry about this one, didn't get highlighted as a conflict

File: apps/dc/dc-purview-shir/dc-purview-shir.yaml (hunk 4 above)
Comment (HarrisonBoyleThomas) [reply apps/dc/dc-purview-shir/dc-purview-shir.yaml]: Changes made

File: apps/dc/dc-purview-shir/dc-purview-shir.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: dc-purview-shir-deployment
+  namespace: dc
+spec:
+  interval: 5m
+  releaseName: dc-purview-shir
+  values:
```
[the remark this answers is not in the document]
Comment (HarrisonBoyleThomas) [reply apps/dc/dc-purview-shir/dc-purview-shir.yaml:9]: Bit confused by this one, do you mean by adding a new `values` attribute to chart-base, and moving the entire spec for the dc-purview-shir inside there?

[the remark this answers is not in the document]
Comment (HarrisonBoyleThomas) [reply apps/dc/dc-purview-shir/dc-purview-shir.yaml]: Like this? https://github.com/hmcts/sds-flux-config/pull/3778/commits/709f03434d1cafe226bc369a80736db1b4ebcfbc

[the remark this answers is not in the document]
Comment (HarrisonBoyleThomas) [reply apps/dc/dc-purview-shir/dc-purview-shir.yaml:8]: Actually I think i've found an example of this in cnp-flux-config, made the change

File: apps/dc/dc-purview-shir/dev.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: dc-purview-shir-deployment
+  namespace: dc
+spec:
+  releaseName: dc-purview-shir
+  values:
+    image: sdshmctspublic.azurecr.io/dc/dc-purview-integration-runtime:dev-d3daaf9-20231103094138 #{"$imagepolicy": "flux-system:dc-purview-shir-dev"}
```
[the remark this answers is not in the document]
Comment (HarrisonBoyleThomas) [reply apps/dc/dc-purview-shir/dev.yaml]: Changes made

File: apps/dc/dc-purview-shir/dev.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+    replicaCount: 4
+    memoryLimits: '4096Mi'
+    environment: "dev"
+    resourceGroup: "dc-dev-rg"
+    subscriptionId: "867a878b-cb68-4de5-9741-361ac9e178b6"
+    tenantId: "531ff96d-0ae9-462a-8d2d-bec7c0b42082"
+    managedIdentityClientId: "e9269bef-5b35-454b-b302-8ed1ba014375"
```
[the remark this answers is not in the document]
Comment (HarrisonBoyleThomas) [reply apps/dc/dc-purview-shir/dev.yaml]: Changes made

**Review by timja — APPROVED** [review]

**Pull request merged**